.tree-sitter-lint/tree-sitter-lint-local/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+use std::{path::Path, sync::Arc};
+
+use tree_sitter_lint::{
+    clap::Parser, tree_sitter::Tree, tree_sitter_grep::RopeOrSlice, Args, Config, Plugin, Rule,
+    ViolationWithContext,
+};
+
+pub fn run_and_output() {
+    tree_sitter_lint::run_and_output(args_to_config(Args::parse()));
+}
+
+pub fn run_for_slice<'a>(
+    file_contents: impl Into<RopeOrSlice<'a>>,
```

Comment:
Just "bubbling this up", maybe this would/could also become the more generic `impl TextProvider + Parseable` or whatever (once the underlying `tree_sitter_lint::run_for_slice()` -> `tree_sitter_grep::whatever_its_called()` take that)?